include/ppx/metrics.h

```diff
@@ -285,9 +386,24 @@ class Manager final
     // Returns whether a run is active.
     bool HasActiveRun() const;
 
+    // Allocate a MetricID.
+    MetricID AllocateID() { return EnsureAllocateID(kInvalidMetricID); }
+
     // Adds a metric to the current run. A run must be started to add a metric.
     // Failure to add a metric returns kInvalidMetricID.
-    MetricID AddMetric(const MetricMetadata& metadata);
+    // Optionally bind the metric to an existing metricID.
+    MetricID AddMetric(
+        const MetricMetadata& metadata,
+        MetricID              metricID = kInvalidMetricID);
```

Comment:
What about `std::optional<MetricID> metricId = std::nullopt`?

Reply:
`std::optional<MetricID>` have two null value, i.e. `std::nullopt` and `kInvalidMetricID`.
In which case I'd rather have `MetricID AddMetric(MetricMetadata)` and `bool BindMetric(MetricID,MetricMetadata)`

Reply:
Yeah.. I agree wince we have 2 null values, optional is not a good idea. Don't like those magic numbers for null when we have alternatives, but not for this PR I suppose.
Let's disregard my comment!